tck/src/main/java/org/eclipse/microprofile/lra/tck/TckCancelOnTests.java

```diff
@@ -0,0 +1,198 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.eclipse.microprofile.lra.tck;
+
+import static org.eclipse.microprofile.lra.tck.participant.api.LraCancelOnController.LRA_CANCEL_ON_CONTROLLER_PATH;
+import static org.junit.Assert.assertEquals;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.util.logging.Logger;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+
+import org.eclipse.microprofile.lra.tck.participant.api.LraCancelOnController;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+import org.junit.runner.RunWith;
+
+@RunWith(Arquillian.class)
+@RunAsClient
+public class TckCancelOnTests {
+    private static final Logger LOGGER = Logger.getLogger(TckCancelOnTests.class.getName());
+
+
+    @Rule public TestName testName = new TestName();
+
+    private static Client tckSuiteClient;
+
+    private int beforeCompletedCount;
+    private int beforeCompensatedCount;
+
+    @Deployment(name = "tcktests-cancelon", managed = true, testable = true)
+    public static WebArchive deploy() {
+        String archiveName = TckCancelOnTests.class.getSimpleName().toLowerCase();
+        return ShrinkWrap
+            .create(WebArchive.class, archiveName + ".war")
+            .addPackages(true, "org.eclipse.microprofile.lra.tck")
+            .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml");
+    }
+
+    @BeforeClass
+    public static void beforeClass() {
+        tckSuiteClient = ClientBuilder.newClient();
+    }
+
+    @AfterClass
+    public static void afterClass() {
+        if(tckSuiteClient != null) {
+            tckSuiteClient.close();
+        }
+    }
+
+    @Before
+    public void before() {
+        LOGGER.info("Running test: " + testName.getMethodName());
+        this.beforeCompletedCount = getCompletedCount();
+        this.beforeCompensatedCount = getCompensateCount();
+    }
+
+    private WebTarget getSuiteTarget() {
+        String tckSuiteBaseUrl = System.getProperty("lra.tck.base.url");
```

Comment:
why are you getting this directly when you are injecting this config property in `LraTckConfigBean`?

Reply:
this is a think I'm not convinced I did in the best way. The `TckCancelOnTests` are written to be run on client (`@RunAsClient`). It means it runs in the surefire/failsafe JVM and there is no container which would manage the property injection. I think the MicroProfile config projects can be used as standalone libraries but the configuration could be a bit difficult. Then I took this approach for the tests on the client.

Reply:
Thanks for the clarification - it seems reasonable to me

Reply:
Thanks. This is ok with me. We can take look after merge of this PR. But maybe we could add a note to the setup document?

Reply:
@xstefank good point. I provided the default value as defined in the document and added a sentence mentioning this.